R/codecov.R

```diff
@@ -0,0 +1,190 @@
+#' Run covr on a package and upload the result to codecov.io
+#' @param path file path to the package
+#' @param Codecov url (change for Enterprise)
+#' @param ... additional arguments passed to \code{\link{package_coverage}}
+#' @export
+codecov <- function(path = ".", base_url = 'https://codecov.io', ...) {
+  
+  # -------
+  # Jenkins
+  # -------
+  if (Sys.getenv('JENKINS_URL')) {
+    # https://wiki.jenkins-ci.org/display/JENKINS/Building+a+software+project
+    # path <- Sys.getenv("WORKSPACE")
+    codecov_url <- paste(base_url, "/upload/v2?service=jenkins", 
+                         "&branch=", Sys.getenv('GIT_BRANCH'), 
+                         "&commit=", Sys.getenv('GIT_COMMIT'),
+                         "&build=", Sys.getenv('BUILD_NUMBER'),
+                         "&build_url=", Sys.getenv('BUILD_URL'), sep="")
+
+  # ---------
+  # Travis CI
+  # ---------
+  } else if (Sys.getenv('CI') == "true" && Sys.getenv('TRAVIS') == "true") {
+    # http://docs.travis-ci.com/user/ci-environment/#Environment-variables
+    # path <- Sys.getenv('TRAVIS_BUILD_DIR')
+    codecov_url <- paste(base_url, "/upload/v2?service=travis-org",
+                         "&branch=", Sys.getenv('TRAVIS_BRANCH'),
+                         "&build=", Sys.getenv('TRAVIS_JOB_NUMBER'),
+                         "&pull_request=", ifelse(Sys.getenv('TRAVIS_PULL_REQUEST')!='false', Sys.getenv('TRAVIS_PULL_REQUEST'), ''),
+                         "&travis_job_id=", Sys.getenv('TRAVIS_JOB_ID'),
+                         "&owner=", strsplit(Sys.getenv('TRAVIS_REPO_SLUG'), '/')[0],
+                         "&repo=", strsplit(Sys.getenv('TRAVIS_REPO_SLUG'), '/')[1],
+                         "&commit=", Sys.getenv('TRAVIS_COMMIT'), sep="")
+  # --------
+  # Codeship
+  # --------
+  } else if (Sys.getenv('CI') == "true" && Sys.getenv('CI_NAME') == 'codeship') {
+    # https://www.codeship.io/documentation/continuous-integration/set-environment-variables/
+    codecov_url <- paste(base_url, "/upload/v2?service=codeship",
+                         "&branch=", Sys.getenv('CI_BRANCH'),
+                         "&build=", Sys.getenv('CI_BUILD_NUMBER'),
+                         "&build_url=", Sys.getenv('CI_BUILD_URL'),
+                         "&commit=", Sys.getenv('CI_COMMIT_ID')), sep="")
```

Comment:
There's an extra closing parenthesis before the `sep` -- and the same issue also occurs at a few other lines below.